### PR TITLE
Part of #18714 and #17670:set-approval-for-all-warnings

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation
Replaces deprecated Typography and Button components in `ui\components\app\set-approval-for-all-warning\set-approval-for-all-warning.js` as well as fixes strong tag issue in Text component

Part of

- https://github.com/MetaMask/metamask-extension/issues/18714
- https://github.com/MetaMask/metamask-extension/issues/17670
- https://github.com/MetaMask/metamask-extension/issues/18896

### Before

Screenshot 2023-05-15 at 9 21 41 AM

### After

Screenshot 2023-05-15 at 9 20 52 AM

## Manual Testing Steps

- Pull this branch
- Run `yarn storybook`
- Check the `SetApproveForAllWarning` story

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
Hey @georgewrmarshall could you please review this PR?
I made the required changes, there has been no visual changes to the stories but theres a error in the line which required linting changes after replacing `<Typography>` with `<Text>` (refer the image below)
image
changed to 
image

It shows no problems in the code, all ci tests pass aswell

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Code coverage failing is because of linting formatting change

File: ui/components/app/set-approval-for-all-warning/set-approval-for-all-warning.js [hunk 1]
```diff
@@ -65,12 +65,13 @@ const SetApproveForAllWarning = ({
           name={IconName.Danger}
           className="set-approval-for-all-warning__content__header__warning-icon"
         />
-        <Typography
-          variant={TypographyVariant.H4}
-          fontWeight={FONT_WEIGHT.BOLD}
+        <Text
+          variant={TextVariant.headingSm}
+          as="h4"
+          fontWeight={FontWeight.Bold}
```
[the remark this answers is not in the document]
Comment (dhruvv173) [reply ui/components/app/set-approval-for-all-warning/set-approval-for-all-warning.js]: I made the suggested changes but the same tests are failing again :/
Sorry for the inconvenience.

File: ui/components/app/set-approval-for-all-warning/index.scss [hunk 2]
```diff
@@ -13,10 +13,6 @@
     &__account {
       border-bottom: 1px solid var(--color-border-muted);
     }
-
-    &__bold {
-      color: var(--color-text-default);
-    }
```
Comment (georgewrmarshall) [inline ui/components/app/set-approval-for-all-warning/index.scss:19]: Removed styles and made all text text/default color

File: ui/components/component-library/text/text.scss [hunk 3]
```diff
@@ -33,7 +33,7 @@ $text-variants: (
   color: var(--color-text-default);
   font-family: var(--font-family-sans);
 
-  > strong {
+  strong {
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.scss:36]: Fixes issue where only the direct strong tag child would be bold font-weight. That meant if it was wrapped in a `span` which many of the translations are they would not receive the styles properly.

**Review by georgewrmarshall — APPROVED** [review]
Hey @dhruvv173, by comparing the before and after screenshots in your PR description I was able to spot some issues with our `Text` component and strong tag. I went to the liberty of updating it in your PR I hope you don't mind. I've also updated some more deprecated components with the new ones as well as updated the PR description and before/after screenshots to the most recent. Thanks so much for your contribution!!

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, by comparing the before and after screenshots in your PR description I was able to spot some issues with our `Text` component and strong tag. I went to the liberty of updating it in your PR I hope you don't mind. I've also updated some more deprecated components with the new ones as well as updated the PR description and before/after screenshots to the most recent. Thanks so much for your contribution!!

Hey @georgewrmarshall, apologies for the mistakes from my end and thanks a lot for your input and help. You've been very helpful and polite while handling issues, really appreciate it 🙏 ❤️

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, there were no mistakes on your end. These were just further improvements that could be made that weren't outlined in the original issue. It was your added screenshots that helped with these improvements so thanks for making sure your PR description was detailed. It makes a lot of difference! 👍

**Review by garrettbear — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Pull request merged**